src/CFamily.UnitTests/CompilationDatabase/CompilationDatabaseLocatorTests.cs

```diff
@@ -0,0 +1,204 @@
+        [DataRow(false)]
+        public void Locate_NoCMakeSettingsFile_ReturnsDefaultLocationIfItExists(bool defaultFileExists)
+        {
+            var activeConfiguration = CompilationDatabaseLocator.DefaultConfiguration;
+            var defaultLocation = GetDefaultDatabaseFileLocation(activeConfiguration);
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.File.Exists(It.IsAny<string>())).Returns(false);
+            fileSystem.Setup(x => x.File.Exists(defaultLocation)).Returns(defaultFileExists);
+
+            var testSubject = CreateTestSubject(RootDirectory, fileSystem.Object);
+
+            var result = testSubject.Locate();
+
+            if (defaultFileExists)
+            {
+                result.Should().Be(defaultLocation);
+            }
+            else
+            {
+                result.Should().BeNull();
+            }
+
+            fileSystem.Verify(x=> x.File.Exists(defaultLocation), Times.Once);
+        }
+
+        [TestMethod]
+        public void Locate_HasCMakeSettingsFile_ActiveConfigurationDoesNotExist_Null()
+        {
+            var cmakeSettingsLocation = GetCmakeSettingsLocation(RootDirectory);
+
+            var fileSystem = new Mock<IFileSystem>();
+            SetupCMakeSettingsFileExists(fileSystem, cmakeSettingsLocation, new CMakeSettings());
+
+            var testSubject = CreateTestSubject(RootDirectory, fileSystem.Object);
+
+            var result = testSubject.Locate();
+
+            result.Should().BeNull();
+
+            fileSystem.Verify(x=> x.File.ReadAllText(cmakeSettingsLocation), Times.Once);
+        }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void Locate_HasCMakeSettingsFile_ReturnsConfiguredPathIfItExists(bool configuredPathExists)
+        {
+            var activeConfiguration = CompilationDatabaseLocator.DefaultConfiguration;
+            var cmakeSettings = CreateCMakeSettings(activeConfiguration, "folder");
+            var cmakeSettingsLocation = GetCmakeSettingsLocation(RootDirectory);
+
+            var fileSystem = new Mock<IFileSystem>();
+            SetupCMakeSettingsFileExists(fileSystem, cmakeSettingsLocation, cmakeSettings);
+
+            var compilationDatabaseFullLocation = Path.GetFullPath(
+                Path.Combine("folder", CompilationDatabaseLocator.CompilationDatabaseFileName));
+
+            fileSystem.Setup(x => x.File.Exists(compilationDatabaseFullLocation)).Returns(configuredPathExists);
+
+            var testSubject = CreateTestSubject(RootDirectory, fileSystem.Object);
+
+            var result = testSubject.Locate();
+
+            if (configuredPathExists)
+            {
+                result.Should().Be(compilationDatabaseFullLocation);
+            }
+            else
+            {
+                result.Should().BeNull();
+            }
+
+            fileSystem.Verify(x => x.File.Exists(compilationDatabaseFullLocation), Times.Once);
```

Comment:
In general, it would be worth asserting that some of the expected data is logged e.g. the missing build config name, the missing file path etc.